### ISSUE TITLE
Prompt Cache loading failure

### ISSUE THREAD

**Issue by mwt11a** [ggerganov/llama.cpp#6705]
--prompt-cache seems to be broken. Saving of a new cache seems fine. But on trying to load that cache, get:

llama_get_logits_ith: invalid logits id -1, reason: negative index out of range [0, 0)
Segmentation fault (core dumped)

Have tried with 7B Mistral and 35B Command-r models.

Commit: e3c337d87ca650972105a51c6ce302dd236c07ad  seems to be the cause. 

Cache loading is working fine on the commit before that one.

**Comment by compilade** [comment]
This has been fixed in #6687, previously reported in #6679. Does using the most recent version fix the problem?

**Comment by mwt11a** [comment]
Yes, sorry, is fixed in main. 

Noticed the issue yesterday and thought I had updated again this morning, but realized I hadn't. Had spent the last hour checking the various commits to find what one was the cause, but had forgot to check if fixed in most recent.
